### PR TITLE
Show packet "Friendly Name" in send / recv log if known

### PR DESCRIPTION
**Pull request by cyberkitsune**
This has been bugging me for 5 years.

Code review requested.

### CODE REVIEW

File: PolarisServer/Client.cs [hunk 1]
```diff
@@ -204,7 +224,7 @@ public void SendPacket(Packet packet)
             Array.Copy(data, position, packet, 0, size);
             LogPacket(true, typeA, typeB, flags1, flags2, packet);
 
-            var handler = PacketHandlers.GetHandlerFor(typeA, typeB);
+
```
Comment (michael-fadely) [inline PolarisServer/Client.cs:227]: Empty line

File: PolarisServer/Packets/Packet.cs [hunk 2]
```diff
@@ -6,5 +6,6 @@ public abstract class Packet
     {
         public abstract byte[] Build();
         public abstract PacketHeader GetHeader();
+
```
Comment (michael-fadely) [inline PolarisServer/Packets/Packet.cs:9]: Added empty line in otherwise unchanged file.

File: PolarisServer/Client.cs [hunk 3]
```diff
@@ -130,14 +130,24 @@ private void HandleConnectionLost()
...
         }
 
-        public void SendPacket(byte[] blob)
+        public void SendPacket(byte[] blob, string name = null)
         {
             var typeA = blob[4];
             var typeB = blob[5];
             var flags1 = blob[6];
             var flags2 = blob[7];
+            string sendName = "";
```
Comment (michael-fadely) [inline PolarisServer/Client.cs:139]: Assigned value not used in any code path. Also recommend preferring `string.Empty`

File: PolarisServer/Client.cs [hunk 4]
```diff
@@ -130,14 +130,24 @@ private void HandleConnectionLost()
...
 
-            Logger.Write("[<--] Packet {0:X}-{1:X} (flags {2}) ({3} bytes)", typeA, typeB, (PacketFlags)flags1,
+            if (name != null)
+            {
+                sendName = string.Format("{0} ({1:X}-{2:X})", name, typeA, typeB);
```
Comment (michael-fadely) [inline PolarisServer/Client.cs:143]: String interpolation recommended. e.g.:
```cs
sendName = $"{name} ({typeA:X}-{typeB:X})";
```

File: PolarisServer/Client.cs [hunk 5]
```diff
@@ -130,14 +130,24 @@ private void HandleConnectionLost()
...
+            }
+            else
+            {
+                sendName = string.Format("{0:X}-{1:X}", typeA, typeB);
```
Comment (michael-fadely) [inline PolarisServer/Client.cs:147]: String interpolation recommended.

File: PolarisServer/Client.cs [hunk 6]
```diff
@@ -130,14 +130,24 @@ private void HandleConnectionLost()
...
+            }
+
+            Logger.Write("[<--] Packet {0} (flags {1}) ({2} bytes)", sendName, (PacketFlags)flags1,
                 blob.Length);
```
Comment (michael-fadely) [inline PolarisServer/Client.cs:151]: String interpolation recommended.

File: PolarisServer/Client.cs [hunk 7]
```diff
@@ -177,19 +187,29 @@ public void SendPacket(byte typeA, byte typeB, byte flags, byte[] data)
...
-            SendPacket(h.Type, h.Subtype, h.Flags1, packet.Build());
+            SendPacket(h.Type, h.Subtype, h.Flags1, packet.Build(), packet.GetType().Name);
         }
 
         private void HandlePacket(byte typeA, byte typeB, byte flags1, byte flags2, byte[] data, uint position,
             uint size)
         {
-            Logger.Write("[-->] Packet {0:X}-{1:X} (flags {2}) ({3} bytes)", typeA, typeB, (PacketFlags)flags1, size + 8);
+            var handler = PacketHandlers.GetHandlerFor(typeA, typeB);
+            string packetName = "";
```
Comment (michael-fadely) [inline PolarisServer/Client.cs:203]: Assigned value not used in any code path. Also recommend preferring `string.Empty`

File: PolarisServer/Client.cs [hunk 8]
```diff
@@ -177,19 +187,29 @@ public void SendPacket(byte typeA, byte typeB, byte flags, byte[] data)
...
+            if (handler != null)
+            {
+                packetName = string.Format("{0} ({1:X}-{2:X})", handler.GetType().Name, typeA, typeB);
```
Comment (michael-fadely) [inline PolarisServer/Client.cs:206]: String interpolation recommended.

File: PolarisServer/Client.cs [hunk 9]
```diff
@@ -177,19 +187,29 @@ public void SendPacket(byte typeA, byte typeB, byte flags, byte[] data)
...
+            }
+            else
+            {
+                packetName = string.Format("{0:X}-{1:X}", typeA, typeB);
```
Comment (michael-fadely) [inline PolarisServer/Client.cs:210]: String interpolation recommended.

File: PolarisServer/Client.cs [hunk 10]
```diff
@@ -177,19 +187,29 @@ public void SendPacket(byte typeA, byte typeB, byte flags, byte[] data)
...
+            }
+            Logger.Write("[-->] Packet {0} (flags {1}) ({2} bytes)", packetName, (PacketFlags)flags1, size + 8);
```
Comment (michael-fadely) [inline PolarisServer/Client.cs:212]: String interpolation recommended.

**Comment by cyberkitsune** [comment]
LET'S GOOOOO

**Pull request merged**